### PR TITLE
[OPENY-82] Facility CT decoupling

### PR DESCRIPTION
**Pull request by anprok**
"OpenY Facility" depends on these modules at this moment
openy_prgf_loc_finder, openy, openy_demo_tfacility, openy_demo_nfacility

## Steps for review

- [x]  login as admin
- [x]   go to /admin/content?title=&type=facility&status=All&langcode=All
- [x]   Check that some facility nodes exist in this list
- [x]   go to /admin/modules/uninstall
- [x]   uninstall "OpenY Demo Taxonomy Facility" module
- [x]   uninstall "OpenY Demo Node Facility" module
- [x]   uninstall "OpenY Paragraph Location Finder" module
- [x]   uninstal "OpenY Facility" module
- [x]   return to /admin/content?title=&type=facility&status=All&langcode=All
- [x]   check that facility nodes was removed
- [x]   go to /admin/structure/types
- [x]   check that "Facility" not exist in this list
- [x]   go to /admin/modules
- [x]  install "OpenY Facility" module
- [x]  check that all components related to "OpenY Facility" module was restored

## General checks
- [ ] All coding styles are fulfilled and there are no any issues reported by CodeSniffer CI. <br/>CI code sniffer errors
- [ ] All tests are running and there are no failed tests reported by CI. <br/>Behat test results
- [ ] [Documentation](https://github.com/ymcatwincities/openy/tree/8.x-1.x/docs) has been updated according to PR changes.
- [ ] [Steps for review](https://github.com/ymcatwincities/openy/pull/94#issue-204580200) have been provided according to PR changes. <br/>Steps for review
- [ ] Make sure you've provided all necessary hook\_update\_N to [support upgrade path](https://github.com/ymcatwincities/openy/blob/8.x-1.x/docs/Development/Upgrade%20path.md).
- [ ] Make sure your git email is associated with account on drupal.org, otherwise you won't get commits there. <br/>drupal.org email
- [ ] If you would like to get credits on drupal.org, [check documentation](https://github.com/ymcatwincities/openy/blob/8.x-1.x/docs/Development/Contributing.md#drupalorg-credits).

Thank you for your contribution!

### CODE REVIEW

**Comment by gundevel** [comment]
Can one of the admins verify this patch? Use "o+k to test" or ''t+est this please" for manual build execution.

**Review by ddrozdik — APPROVED** [review]

**Comment by AlexNetman** [comment]
ok to test

**Comment by ddrozdik** [comment]
retest this please

**Comment by gundevel** [comment]
Build comment file: 
--------------

 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation - Rose theme | http://openy.cibox.tools/build631 |
| Fresh OpenY installation - Lily theme | http://lily.openy.cibox.tools/build631 |
| Upgraded(upgrade path) installation | http://upgrade.openy.cibox.tools/build631 |
| Installation process | http://install.openy.cibox.tools/build631/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/631/

**Comment by gundevel** [comment]
Acessibility Sniffer: front page checking WCAG2AA http://openy.cibox.tools/build631/frontWCAG2AAhtmlcs.txt
Acessibility Sniffer: join page checking WCAG2AA http://openy.cibox.tools/build631/joinWCAG2AAhtmlcs.txt
Acessibility Sniffer: locations page checking WCAG2AA http://openy.cibox.tools/build631/locationsWCAG2AAhtmlcs.txt
Acessibility Sniffer: schedules page checking WCAG2AA http://openy.cibox.tools/build631/schedulesWCAG2AAhtmlcs.txt
Acessibility Sniffer: blog page checking WCAG2AA http://openy.cibox.tools/build631/blogWCAG2AAhtmlcs.txt

**Comment by ddrozdik** [comment]
@anprok currently, `OpenY Facility` module has a lot of dependencies to content types that should not be in the list of direct dependencies, for example `OpenY Session`. First of all, we have to resolve these dependencies. See how we did this before #1118

**Comment by ddrozdik** [comment]
@anprok there is no upgrade path for the modified configs. Please add upgrade path.

File: modules/openy_features/openy_location/modules/openy_loc_facility/config/install/core.entity_form_display.node.facility.default.yml [hunk 1]
```diff
@@ -17,6 +17,7 @@ dependencies:
     - node.type.facility
   module:
     - address
+    - content_moderation
```
Comment (podarok) [inline modules/openy_features/openy_location/modules/openy_loc_facility/config/install/core.entity_form_display.node.facility.default.yml]: No need to add content moderation for facility. This is the super rare case when Ys adding facilities.

File: modules/openy_features/openy_location/modules/openy_loc_facility/config/install/core.entity_form_display.node.facility.default.yml [hunk 2]
```diff
@@ -229,6 +236,14 @@ content:
     region: content
     settings: {  }
     third_party_settings: {  }
+  redirect:
+    type: string_textfield
+    weight: 30
+    region: content
+    settings:
+      size: 60
+      placeholder: ''
```
Comment (podarok) [inline modules/openy_features/openy_location/modules/openy_loc_facility/config/install/core.entity_form_display.node.facility.default.yml]: Could we have placeholder added bu default?

File: modules/openy_features/openy_location/modules/openy_loc_facility/config/install/core.entity_view_display.node.facility.default.yml [hunk 3]
```diff
@@ -28,6 +28,14 @@ targetEntityType: node
 bundle: facility
 mode: default
 content:
+  addthis:
```
Comment (podarok) [inline modules/openy_features/openy_location/modules/openy_loc_facility/config/install/core.entity_view_display.node.facility.default.yml]: I don't like dependency from addthis here
Could we remove it?
add this should be on blogs, news, probably Landing Pages. But Facility  - no need

File: modules/openy_features/openy_location/modules/openy_loc_facility/config/install/field.field.node.facility.field_facility_loc.yml [hunk 4]
```diff
@@ -3,25 +3,31 @@ status: true
 dependencies:
   config:
     - field.storage.node.field_facility_loc
-    - node.type.branch
     - node.type.facility
+  module:
+    - datalayer
+third_party_settings:
+  datalayer:
+    expose: 0
```
Comment (podarok) [inline modules/openy_features/openy_location/modules/openy_loc_facility/config/install/field.field.node.facility.field_facility_loc.yml]: we aren't exposing this field, let's remove dependency from dataLayer

**Review by podarok — CHANGES_REQUESTED** [review]
Needs work

**Review by ddrozdik — APPROVED** [review]

**Comment by ddrozdik** [comment]
retest this please

**Comment by bjwilkins** [comment]
Build comment file: 
--------------

 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation - Rose theme | http://openy.cibox.tools/build636 |
| Fresh OpenY installation - Lily theme | http://lily.openy.cibox.tools/build636 |
| Upgraded(upgrade path) installation | http://upgrade.openy.cibox.tools/build636 |
| Installation process | http://install.openy.cibox.tools/build636/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/636/

**Review by podarok — APPROVED** [review]

**Pull request merged**